controllers/configurationpolicy_utils.go

```diff
@@ -454,32 +444,44 @@ func createStatus(
 			}
 		}
 
-		if !stringInSlice(idStr, nameList) {
-			nameList = append(nameList, idStr)
-		}
+		foundIdentifiers[idStr] = true
 	}
 
-	names := strings.Join(nameList, "; ")
+	niceNames := sortAndJoinKeys(foundIdentifiers, "; ")
 
 	var reason, msg string
 
 	if objShouldExist {
 		if compliant {
 			reason = "K8s `must have` object already exists"
-			msg = fmt.Sprintf("%v %v as specified, therefore this Object template is compliant", kind, names)
+			msg = fmt.Sprintf("%v %v as specified, therefore this Object template is compliant", kind, niceNames)
 		} else {
 			reason = "K8s does not have a `must have` object"
-			msg = fmt.Sprintf("%v not found: %v", kind, names)
+			msg = fmt.Sprintf("%v not found: %v", kind, niceNames)
 		}
 	} else {
 		if compliant {
 			reason = "K8s `must not have` object already missing"
-			msg = fmt.Sprintf("%v %v missing as expected, therefore this Object template is compliant", kind, names)
+			msg = fmt.Sprintf("%v %v missing as expected, therefore this Object template is compliant", kind, niceNames)
 		} else {
 			reason = "K8s has a `must not have` object"
-			msg = fmt.Sprintf("%v found: %v", kind, names)
+			msg = fmt.Sprintf("%v found: %v", kind, niceNames)
 		}
 	}
 
 	return addConditionToStatus(plc, indx, compliant, reason, msg)
 }
+
+func sortAndJoinKeys(m map[string]bool, sep string) string {
+	keys := make([]string, len(m))
+	i := 0
+
+	for key := range m {
+		keys[i] = key
+		i++
+	}
+
+	sort.Strings(keys)
```

Comment:
Optional: In case you wanted to be brave and use an insertion sort, you could do this:
https://stackoverflow.com/a/55460931

I do think it's overkill for our needs though!